src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,142 @@
+        }
+
+        array[index] = value;
+        arraySize++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        System.out.println("public void addAll");
+        if (list.isEmpty()) {
+            return;
+        }
+
+        int newSize = arraySize + list.size();
 
+        T[] newArray = (T[]) new Object[newSize];
+        System.arraycopy(array, 0, newArray, 0, arraySize);
+        array = newArray;
+
+        for (int i = 0; i < list.size(); i++) {
+            array[arraySize++] = list.get(i);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= 0 && index < arraySize) {
+            return array[index];
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Can't find this index in this array");
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= 0 && index < arraySize) {
+            array[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index >= 0 && index < arraySize) {
+            final T removedElement = array[index];
+            for (int i = index; i < arraySize - 1; i++) {
+                array[i] = array[i + 1];
+            }
+            arraySize--;
+            array[arraySize] = null;
+            return removedElement;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        if (element == null) {
+            for (int i = 0; i < arraySize; i++) {
+                if (array[i] == null) {
+                    final T removedElement = array[i];
+                    for (int j = i; j < arraySize - 1; j++) {
+                        array[j] = array[j + 1];
+                    }
+                    arraySize--;
+                    array[arraySize] = null;
+                    return removedElement;
+                }
+            }
+        } else {
+            for (int i = 0; i < arraySize; i++) {
+                if (array[i] != null && array[i].equals(element)) {
+                    final T removedElement = array[i];
+                    for (int j = i; j < arraySize - 1; j++) {
+                        array[j] = array[j + 1];
+                    }
+                    arraySize--;
+                    array[arraySize] = null;
+                    return removedElement;
+                }
+            }
+        }//
+
+        throw new NoSuchElementException("Element not found");
     }
 
     @Override
     public int size() {
-        return 0;
+        System.out.println("public int size()");
+        return arraySize;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        System.out.println("public boolean isEmpty()");
```

Comment:
Remove the debug output 'System.out.println()' from the 'isEmpty' method. It's unnecessary and can clutter the console output.